### PR TITLE
Process from template

### PR DESCRIPTION
**Pull request by chrizzor**
This pr adds the template to create processes from a template.
We put the search functionality on hold because of the new import module.
So the search isn't working yet.

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 1]
```diff
@@ -2013,4 +2025,26 @@ public boolean isCalendarButtonShowing() {
...
+            if (!callerViewId.isEmpty()
+                    && (callerViewId.contains("projects.jsf"))) {
+                return PROCESS_FROM_TEMPLATE_PATH + "?" + REDIRECT_PARAMETER;
+            } else {
+                return PROCESS_FROM_TEMPLATE_PATH_OLD + "?" + REDIRECT_PARAMETER;
+            }
+        } catch (NullPointerException e) {
+            // This NPE gets thrown - and therefore must be caught - when "ProzessverwaltungForm" is
+            // used from it's integration test
+            // class "ProzessverwaltungFormIT", where no "FacesContext" is available!
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: On copy and pasting code from other classes, please adjust the comments. You added this to class `ProzesskopieForm` but your comment is about class `ProzessverwaltungFormIT` two times.

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/processFromTemplate/processDataEdit.xhtml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                    <f:selectItems value="#{ProzesskopieForm.allDoctypes}" var="step" itemLabel="#{step.localizedLabel}"
+                                   itemValue="#{step.title}"/>
+                </p:selectOneMenu>
+            </div>
+
+            <div>
+                <!-- Preferences -->
+                <p:outputLabel for="Regelsatz" rendered="#{ProzesskopieForm.standardFields.preferences}" value="#{msgs.regelsatz}"/>
+                <h:panelGroup rendered="#{ProzesskopieForm.standardFields.preferences}">
+                    <p:selectOneMenu id="Regelsatz" value="#{ProzesskopieForm.prozessKopie.ruleset}" converter="#{rulesetConverter}"
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/WEB-INF/templates/includes/processFromTemplate/processDataEdit.xhtml]: Did this really work? I can not see a selectable list of rulesets.

**Comment by henning-gerhardt** [comment]
general functionality:
- switching document type did not change the entry fields on additional details tab
- process title generation did not work and with this the fields "Tiff header document name" and "Tiff header image description" did not get filled
- saving did not work - maybe a error message is displayed for less then a second but it is not visible for a user
- a lot of unhandled FacesMessages are created like the following one:
```
Mar 06, 2018 1:50:26 PM org.apache.myfaces.lifecycle.RenderResponseExecutor execute
WARNING: There are some unhandled FacesMessages, this means not every FacesMessage had a chance to be rendered.
These unhandled FacesMessages are: 
- j_id_2h:j_id_2p:j_id_3v:4:j_id_3y: Validation Error: Value is required
```

design / displaying:
- display of digital collections is wrapped as used "box" is very small
- single and multi selection fields on additional details tab are not very well visible
- no visible difference between mandatory and optional fields on additiona details tab

**Comment by solth** [comment]
> general functionality:
> 
>     switching document type did not change the entry fields on additional details tab
>     process title generation did not work and with this the fields "Tiff header document name" and "Tiff header image description" did not get filled
>     saving did not work - maybe a error message is displayed for less then a second but it is not visible for a user
>     a lot of unhandled FacesMessages are created like the following one:
> 
> Mar 06, 2018 1:50:26 PM org.apache.myfaces.lifecycle.RenderResponseExecutor execute
> WARNING: There are some unhandled FacesMessages, this means not every FacesMessage had a chance to be rendered.
> These unhandled FacesMessages are: 
> - j_id_2h:j_id_2p:j_id_3v:4:j_id_3y: Validation Error: Value is required
> 
> design / displaying:
> 
>     display of digital collections is wrapped as used "box" is very small
>     single and multi selection fields on additional details tab are not very well visible
>     no visible difference between mandatory and optional fields on additiona details tab

These issues should now all be resolved.

**Pull request merged**